### ISSUE TITLE
gradient not update

### ISSUE THREAD

**Issue by searchlink** [leemengtaiwan/leemengtaiwan.github.io#7]
hi, 感谢你的博文，非常精彩，收获良多。在文章```attack_on_bert_transfer_learning_in_nlp``,我根据你的示例利用bert微调下游任务，发现loss和acc没有更新。琢磨了很久，不清楚为什么梯度没更新，代码完全复制你的，能帮我指点一下吗？另外想问下实际训练的参数量不应该是线性分类器的参数个数2307吗？按照你的代码，会训练整个模型吗？
```
def get_learnable_params(module):
    return [p for p in module.parameters() if p.requires_grad]
model_params = get_learnable_params(model)
clf_params = get_learnable_params(model.classifier)
print("整个分类模型的参数量：", sum(p.numel() for p in model_params))
print("线性分类器的参数量：", sum(p.numel() for p in clf_params))

整个分类模型的参数量： 102269955
线性分类器的参数量： 2307
开始训练：
correct: 1805 total: 2657
[epoch 1] loss: 40.695, acc: 0.679
correct: 1805 total: 2657
[epoch 2] loss: 31.688, acc: 0.679
correct: 1805 total: 2657
[epoch 3] loss: 30.562, acc: 0.679
correct: 1805 total: 2657
[epoch 4] loss: 30.287, acc: 0.679
correct: 1805 total: 2657
[epoch 5] loss: 30.180, acc: 0.679

```

**Comment by searchlink** [comment]
已经解决！感谢~

**Comment by ShallTearchen** [comment]
@searchlink 请问一下您有没有遇到过模型在所有样例上的预测都是同一类的情况，损失在下降，但是预测结果完全不变
